src/app/components/SearchResults/FilterYears.jsx

```diff
@@ -16,62 +17,84 @@ const getYearsFilter = (searchQuery) => {
 class FilterYears extends React.Component {
   constructor(props) {
     super(props);
-    this.state = getYearsFilter(props.searchQuery);
+    this.state = { ...getYearsFilter(props.searchQuery), ...{ errorMessage: {}, error: {} } };
     this.onChangeYear = this.onChangeYear.bind(this);
   }
 
   componentWillReceiveProps(nextProps) {
-    this.setState(getYearsFilter(nextProps.searchQuery));
+    this.setState(prevState => ({ ...getYearsFilter(nextProps.searchQuery), ...prevState }));
   }
 
   onChangeYear(e, yearType) {
     const val = e.target.value && Number(e.target.value);
     // TODO: errors control UI
-    // if (yearType === 'start' && this.state.end && val && val > Number(this.state.end)) {
-    //   val = this.state.end;
-    // }
-    // if (yearType === 'end' && this.state.start && val && val < Number(this.state.start)) {
-    //   val = this.state.start;
-    // }
+    const errorMessage = {};
+    const error = {};
+    const errorMessageText = errorMessagesText.invalidDate;
+    if (yearType === 'start') {
+      if (this.state.end && val && val > Number(this.state.end)) {
+        errorMessage[yearType] = errorMessageText;
+        error[yearType] = true;
+      } else {
+        errorMessage[yearType] = '';
+        error[yearType] = false;
+      }
+    } else if (yearType === 'end') {
+      if (this.state.start && val && val < Number(this.state.start)) {
+        errorMessage[yearType] = errorMessageText;
+        error[yearType] = true;
+      } else {
+        errorMessage[yearType] = '';
+        error[yearType] = false;
+      }
+    }
+    this.setState({ errorMessage, error });
     const obj = {};
     obj[yearType] = val;
     this.setState(state => Object.assign({}, state, obj));
-    this.props.onChange({ ...this.state, ...obj });
+    if (!error.end && !error.start) {
+      this.props.onChange({ ...{ start: this.state.start, end: this.state.end }, ...obj });
+      this.props.onError({ errorMsg: '', error: false });
+    } else {
+      this.props.onError({ errorMsg: errorMessageText, error: true });
+    }
   }
 
   render() {
     return (
       <div className="grid-container padding-0">
-        <div className="grid-row">
+        <div className={this.props.className}>
           {Object.keys(yearsType).map(yearType => (
-            <label
+            <TextInput
+              className={this.props.inputClassName}
+              ariaLabel="Search for End Date"
```

Comment:
I think the aria-label should be tied to the `yearType`? It seems to be outputting "End" for both the start and end years.

Reply:
Totally right. Just changed